src/dex/jarvis-v6/types.ts

```diff
@@ -0,0 +1,62 @@
+import { Interface } from '@ethersproject/abi';
+import { Address, Token } from '../../types';
+
+export type PoolState = {
+  priceFeed: ChainLinkPriceFeedState;
+  pool: SynthereumPoolState;
+};
+
+export type ChainLinkPriceFeedState = {
+  usdcPrice: bigint;
+};
+
+export type SynthereumPoolState = {
+  feesPercentage: bigint;
+};
+
+export type JarvisV6Data = {
+  poolAddress: string;
+  swapFunction: JarvisSwapFunctions;
+};
+
+export type PoolConfig = {
+  address: Address;
+  priceFeedPair: string;
+  syntheticToken: Token;
+  collateralToken: Token;
+  chainLink: {
+    interface: Interface;
+    address: Address;
+  };
+};
+
+export type priceFeedData = {
+  interface: Interface;
+  address: Address;
+};
+
+export type DexParams = {
+  priceFeed: priceFeedData;
+  pools: PoolConfig[];
+};
+
+type JarvisV6MintParam = [
+  minNumTokens: string,
+  collateralAmount: string,
+  expiration: string,
+  recipient: string,
+];
+
+type JarvisV6RedeemParam = [
+  numTokens: string,
+  minCollateral: string,
+  expiration: string,
+  recipient: string,
+];
+
+export type JarvisV6Params = JarvisV6MintParam | JarvisV6RedeemParam;
+
+export enum JarvisSwapFunctions {
```

Comment:
I can't find equivalent method of `exchange()`, new protocol doesn't allow to exchange jFiat <> jFiat swaps ?

Reply:
The `exchange()` was just the `mint() + redeem()` behind. We removed the `exhange()` method and will deploy another contract that will handle the `exchange()` things.

Reply:
Got you.
Sorry I meant "new contract" instead of new protocol.